core/src/test/scala/tamer/utils/FakeConsumer.scala

```diff
@@ -22,7 +21,8 @@ sealed class FakeConsumer[IK, IV](
     val inFlight: Queue[(TopicPartition, ProducerRecord[IK, IV])],
     val produced: Map[TopicPartition, Queue[ProducerRecord[IK, IV]]],
     log: LogWriter[Task]
-) extends Service {
+) extends Consumer {
+  val r = new java.util.Random()
```

Comment:
layer?

Reply:
I think this was because I could not override and constrain `R` at the same time but I'll have a second look